test/end2end_test.go

```diff
@@ -3901,6 +3901,36 @@ func (s) TestClientRequestBodyErrorCloseAfterLength(t *testing.T) {
 	}
 }
 
+func testClientInvalidStreamID(t *testing.T, e env) {
+	te := newTest(t, e)
+	ts := &funcServer{streamingInputCall: func(stream testgrpc.TestService_StreamingInputCallServer) error {
+		_, err := stream.Recv()
+		return err
+	}}
+	te.startServer(ts)
+	defer te.tearDown()
+	te.withServerTester(func(st *serverTester) {
+		st.writeHeadersGRPC(2, "/grpc.testing.TestService/StreamingInputCall", true)
+		_, err := st.fr.ReadFrame()
+		if err == nil {
+			t.Fatalf("Error expected when Client StreamID is even %v", err)
+		}
+	})
+}
+
+// Client must always send a streamID in odd numbers according to
+// https://httpwg.org/specs/rfc7540.html#StreamIdentifiers. This test
+// makes sure that the transport throws an error when the client streamID is
+// even.
+func (s) TestClientInvalidStreamID(t *testing.T) {
+	for _, e := range listTestEnv() {
```

Comment:
What's the benefit of running for every `TestEnv` here?

Reply:
No benefit really. Following pattern with other tests.

Reply:
I believe we can do this test without that.